### PR TITLE
Update beacon block proto to match spec

### PR DESCRIPTION
**Pull request by prestonvanloon**

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -145,12 +145,28 @@ message ShardAndCommittee {
...
 message BeaconBlock {
-  repeated bytes ancestor_hashes = 1;
-  uint64 slot = 2;
-  bytes randao_reveal = 3;
-  bytes pow_chain_ref = 4;
-  bytes state_root = 5;
-  google.protobuf.Timestamp timestamp = 6;
-  repeated AggregatedAttestation attestations = 7;
-  repeated SpecialRecord specials = 8;
+  uint64 slot = 1;
```
Comment (rauljordan) [inline proto/beacon/p2p/v1/types.proto]: Can we use ranges here as well?

File: proto/beacon/p2p/v1/types.proto [hunk 2]
```diff
@@ -145,12 +145,28 @@ message ShardAndCommittee {
...
+  bytes randao_reveal_hash32 = 2;
+  bytes candidate_pow_receipt_root_hash32 = 3;
+  // Ancestor hashes represent a skip list of ancestor beacon block hashes.
+  // i'th item is the most recent ancestor whose slot is a multiple of 2^i 
+  // for i in the range of 0 to 31.
+  repeated bytes ancestor_hash32s = 4; 
+  bytes state_root_hash32 = 5;
+  // TODO: attestations should be a list of attestionRecords according to spec
```
Comment (terenc3t) [inline proto/beacon/p2p/v1/types.proto]: This is fine, I think we renamed `AttestationrRecords` to `AggregatedAttestation`

File: proto/beacon/p2p/v1/types.proto (hunk 1 above)
Comment (prestonvanloon) [reply proto/beacon/p2p/v1/types.proto]: Mentioned offline, but we only need ranges for large data protos. For messages that we believe will be small (less than 10 fields) then assigning ID block ranges is not necessary.

File: proto/beacon/p2p/v1/types.proto (hunk 2 above)
Comment (prestonvanloon) [reply proto/beacon/p2p/v1/types.proto]: Discussed offline, we will remove aggregatedAttestation in favor of AttestationRecord

**Comment by prestonvanloon** [comment]
This is part of #781.

**Pull request merged**